src/lib/flow/sol-flow-builder.c

```diff
@@ -116,6 +117,9 @@ sol_flow_builder_del(struct sol_flow_builder *builder)
     }
     sol_ptr_vector_clear(&builder->ports_out_desc);
 
+    free((char *)builder->type_desc.symbol);
+    free((char *)builder->type_desc.options_symbol);
```

Comment:
Agreed. @ricardotk take the chance and use arena in other strdups that will need it. (Make the "builder use arena" patch come first, and then your pathc uses it)